app/models/miq_host_provision_workflow.rb

```diff
@@ -97,6 +97,7 @@ def allowed_storages(_options = {})
     ems = ExtManagementSystem.find_by_id(get_value(@values[:placement_ems_name]))
     return result if ems.nil?
     ems.storages.each do |s|
+      s.ext_management_system = ems
```

Comment:
minor performance nit:  I would put this **after** the line `next unless s.store_type == "NFS"`